src/sentry/integrations/notion/integration.py

```diff
@@ -0,0 +1,246 @@
+you to link Sentry issues to existing databases or pages in Notion.
+Notion is the easy, free, flexible, and visual way to manage your projects
+and organize anything, trusted by millions of people from all over the world.
+"""
+
+FEATURES = [
+    FeatureDescription(
+        """
+        Create and link Sentry issue groups directly to a Notion database in any of your
+        workspaces, providing a quick way to jump from a Sentry bug to tracked ticket!
+        """,
+        IntegrationFeatures.ISSUE_BASIC,
+    ),
+    FeatureDescription(
+        """
+        Link Sentry issues to existing Notion database
+        """,
+        IntegrationFeatures.ISSUE_BASIC,
+    ),
+]
+
+metadata = IntegrationMetadata(
+    description=_(DESCRIPTION.strip()),
+    features=FEATURES,
+    author="The Sentry Team",
+    noun=_("Notion"),
+    issue_url="https://github.com/getsentry/sentry/issues/new?assignees=&labels=Component:%20Integrations&template=bug.yml&title=Notion%20Integration%20Problem",
+    source_url="https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/notion",
+    aspects={},
+)
+
+
+def generate_page_title(page):
+    if page["parent"]["type"] == "database_id":
+        title = page["properties"]["Name"]["title"][0]["plain_text"]
+    else:
+        title = page["properties"]["title"]["title"][0]["plain_text"]
+
+    return title
+
+
+class NotionIntegration(IntegrationInstallation, IssueBasicMixin):
+    def get_client(self):
+        access_token = self.model.metadata["access_token"]
+        return NotionApiClient(api_key=access_token)
+
+    comment_key = "sync_comments"
+    outbound_status_key = "sync_status_outbound"
+    inbound_status_key = "sync_status_inbound"
+    outbound_assignee_key = "sync_assignee_outbound"
+    inbound_assignee_key = "sync_assignee_inbound"
+
+    def get_issue_url(self, key):
+        return f"https://www.notion.so/{key}"
+
+    def get_issue_display_name(self, external_issue):
+        return "{}#{}".format(self.model.metadata["workspace_name"], external_issue.title)
+
+    def get_create_issue_config(self, group, user, **kwargs):
+        kwargs["link_referrer"] = REFERRER
+        fields = super().get_create_issue_config(group, user, **kwargs)
+        notion_client = self.get_client()
+        notion_pages = notion_client.get_databases_list()
+
+        project_field = {
+            "name": "notion_page",
+            "label": "Notion page",
+            "choices": notion_pages,
+            "type": "select",
+            "required": True,
+        }
+
+        return fields + [project_field]
+
+    def get_link_issue_config(self, group, **kwargs):
+        """
+        Return the fields needed for linking to an existing issue
+        """
+        notion_client = self.get_client()
+        notion_pages = notion_client.get_pages()
+        output = [(page["id"], generate_page_title(page)) for page in notion_pages["results"]]
```

Comment:
try/catch -> might raise an error if any of these elements are not there especially in `generate_page_title`